### PR TITLE
sim/crypto: compile mbedtls alternative implementation

### PR DESCRIPTION
**Pull request by ThePassionate**
Signed-off-by: makejian <makejian@xiaomi.com>

### CODE REVIEW

File: boards/sim/sim/sim/configs/crypto/defconfig [hunk 1]
```diff
@@ -68,3 +68,6 @@ CONFIG_START_YEAR=2008
 CONFIG_SYSTEM_DUMPSTACK=y
 CONFIG_SYSTEM_NSH=y
 CONFIG_TESTING_CRYPTO=y
+CONFIG_MBEDTLS_SHA1_ALT=y
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/configs/crypto/defconfig:71]: keep the order and add random too

Comment (acassis) [reply boards/sim/sim/sim/configs/crypto/defconfig:71]: @ThePassionate please use "make savedefconfig" to create the defconfig and avoid human error

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@ThePassionate please fix:
````
In file included from /github/workspace/sources/apps/crypto/mbedtls/mbedtls/include/mbedtls/debug.h:27,
                 from mbedtls/programs/test/selftest.c:52:
Error: /github/workspace/sources/apps/crypto/mbedtls/mbedtls/include/mbedtls/ssl.h:590:5: error: "MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT" is not defined, evaluates to 0 [-Werror=undef]
  590 | #if MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT == 0
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
and
```
/github/workspace/sources/nuttx/tools/testbuild.sh: line 370: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  Normalize sim/crypto
56d55
< CONFIG_MBEDTLS_SELF_TEST=y
Saving the new configuration file
```

**Comment by ThePassionate** [comment]
> @ThePassionate please fix:
> 
> ```
> In file included from /github/workspace/sources/apps/crypto/mbedtls/mbedtls/include/mbedtls/debug.h:27,
>                  from mbedtls/programs/test/selftest.c:52:
> Error: /github/workspace/sources/apps/crypto/mbedtls/mbedtls/include/mbedtls/ssl.h:590:5: error: "MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT" is not defined, evaluates to 0 [-Werror=undef]
>   590 | #if MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT == 0
>       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```
> and
> ```
> /github/workspace/sources/nuttx/tools/testbuild.sh: line 370: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
>   Normalize sim/crypto
> 56d55
> < CONFIG_MBEDTLS_SELF_TEST=y
> Saving the new configuration file
> ```
> ```
https://github.com/apache/nuttx-apps/pull/1995 need to enable MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT because mbedselftest will import parts about ssl

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**